backend/checkmach.ml

```diff
@@ -1077,17 +1076,16 @@ end = struct
       transform_top t ~next ~exn w ("external call to " ^ func) dbg
     | Ispecific s ->
       let effect =
+        let w = create_witnesses t Arch_specific dbg in
         if Debuginfo.assume_zero_alloc dbg
         then
           (* Conservatively assume that operation can return normally. *)
           let nor = V.Safe in
-          let exn = if Arch.operation_can_raise s then V.Safe else V.Bot in
+          let exn = if Arch.operation_can_raise s then V.Top w else V.Bot in
```

Comment:
I am not sure I understand why this
change is needed; could you elaborate?

Reply:
The interpretation of "assume" should be "relaxed", which means `exn` component is allowed to be `Top`.